examples/examples/spread.rs

```diff
@@ -11,10 +11,20 @@ struct FilmDetails {
     director: Option<String>,
 }
 
+#[derive(cynic::QueryFragment, Debug)]
+#[cynic(graphql_type = "Film")]
+struct FilmMoreDetails {
+    release_date: Option<String>,
+}
+
 #[derive(cynic::QueryFragment, Debug)]
 struct Film {
+    id: cynic::Id,
     #[cynic(spread)]
     details: FilmDetails,
+    #[cynic(spread)]
+    more_details: FilmMoreDetails,
+    producers: Option<Vec<Option<String>>>,
```

Comment:
Changing this line to the following does not compile
```rust
    #[cynic(flatten)]
    producers: Vec<String>,
```

Reply:
Ah yeah, I see why that is.  Have a fix, but it's late here.  I'll push it tomorrow

Reply:
Thanks for finding this @liningpan. Thankfully It only affects scalars, but definitely need to fix it somehow.

It's actually harder to fix than I'd hoped, so I've raised this as an issue here https://github.com/obmarg/cynic/issues/775, along with making some failing tests.  Will get to it soon hopefully.